src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+                    // D4 00 05 80 92 92
+                    // − The PN532 is configured to receive frame with passive 424 kbps modulation
+                    // type. The frames are sent back immediately.
+                    // The MSB bit (CRC enable) of CIU_TxMode and CIU_RxMode must be set to 1.
+                    // D4 00 05 00 A2 A2 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                case DiagnoseMode.AttentionRequestTest:
+                    // NumTst = 0x06 : Attention Request Test or ISO/IEC14443-4 card presence detection
+                    // This test can be used by an initiator to ensure that a target/card is still in the field:
+                    // o In case of DEP target, an Attention Request command is sent to the target, and it
+                    // is expected to receive the same answer from the target. In that case, the test is
+                    // declared as successful;
+                    // o In case of ISO/IEC14443-4 card, a R(NACK) block is sent to the card and it is
+                    // expected to receive either a R(ACK) block or the last I-Block. In that case, the test
+                    // is declared as successful (ISO/IEC14443-4 card is still in the RF field).
+                    // In case of no or incorrect response, the Result informs about the status of the transaction
+                    // (refer. to §7.1, p:67)
+                    // − Parameter Length : 0,
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK,
+                    // different from 0x00 is Not OK, Status byte. 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                case DiagnoseMode.SelfAntenaTest:
+                    // NumTst = 0x07 : Self Antenna Test
+                    // This test is used to check the continuity of the transmission paths of the antenna.
+                    // − Parameter Length : 1,
+                    // − Parameter : Threshold used for antenna detection
+                    // (applied in register Andet_Control (@610C), see Error!
+                    // Reference source not found.),
+                    // 7 6 5 4 3 2 1 0
+                    // andet_bot andet_up andet_ithl[1:0] andet_ithh[1:0] andet_en
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK (antenna is detected),
+                    // different from 0x00 is not OK (no antenna is detected). 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                default:
+                    break;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Get or set the timeout when PN532 is in virtual card mode
+        /// </summary>
+        public uint VirtualCardTimeout
+        {
+            get { return _virtualCardTimeout * 50; }
```

Comment:
should this be wrapped in `checked { ... }`?

Reply:
I don't get that point.

Reply:
basically add a protection against overflowing _virtualCardTimeout

Reply:
> basically add a protection against (accidental) overflowing _virtualCardTimeout

I think it's done by this one:
```csharp
if (value / 50 > 0xFF)
     throw new ArgumentException($"{nameof(VirtualCardTimeout)} can't be more than 12750 milliseconds.");
```